scripts/parse_tools/parse_checkers.py

```diff
@@ -12,7 +12,15 @@
 
 ########################################################################
 
-_UNITS_RE = re.compile(r"^[^/@#$%^&*()\|<>\[\]{}?,.]+$")
+_UNITLESS_REGEX                = "1"
+_NON_LEADING_ZERO_NUM          = "[1-9]\d*"
+_CHAR_WITH_UNDERSCORE          = "([a-zA-Z]+_[a-zA-Z]+)+"
+_NEGATIVE_NON_LEADING_ZERO_NUM = f"[-]{_NON_LEADING_ZERO_NUM}"
+_UNIT_EXPONENT                 = f"({_NEGATIVE_NON_LEADING_ZERO_NUM}|{_NON_LEADING_ZERO_NUM})"
+_UNIT_REGEX                    = f"[a-zA-Z]+{_UNIT_EXPONENT}?"
+_UNITS_REGEX                   = f"^({_CHAR_WITH_UNDERSCORE}|{_UNIT_REGEX}(\s{_UNIT_REGEX})*|{_UNITLESS_REGEX})$"
```

Comment:
> The only commit on top of the merge is https://github.com/NCAR/ccpp-framework/commit/6cdd38a032d89d0f79c66f37f0c7172e59f135cd, which is required to parse the metadata in the Unified Forecast System / ccpp-physics (underscores in units, e.g. degree_north).

Is there a plan to resolve this discrepancy?

Reply:
Do you mean bringing this bug fix back to the feature/capgen branch? Oh yes, I am going to do that in the next days by cherry-picking the commit and creating a PR for feature/capgen.

Reply:
Note that this is not a bug in the UFS, units with underscores are in the CCPP standard names dictionary: search for `degree_` in https://github.com/ESCOMP/CCPPStandardNames/blob/main/Metadata-standard-names.md

Reply:
Is `degree_north` in UDUNITS? If so, nevermind.
There are lots of ways to specify location, I don't think we need to introduce variances in the CCPP.

Reply:
The CF conventions use `degrees_north` (note the plural) and since CCPP standard names are extensions of those we need to support the underscore, too. But do not ask me why CCPP standard names, UFS etc use the singular version and not the plural version - maybe @grantfirl knows (either way it's out of scope of this PR, the underscore is the only thing that matters here).

Reply:
FWIW. CF allows for several location descriptions:
![Screenshot 2024-03-12 at 2 39 23 PM](https://github.com/NCAR/ccpp-framework/assets/13949033/72f691dd-8f0e-480a-a526-bb201203dc08)